imgui_widgets.cpp

```diff
@@ -3547,7 +3552,7 @@ bool ImGui::InputTextEx(const char* label, const char* hint, char* buf, int buf_
 
         // Preserve cursor position and undo/redo stack if we come back to same widget
         // FIXME: For non-readonly widgets we might be able to require that TextAIsValid && TextA == buf ? (untested) and discard undo stack if user buffer has changed.
-        const bool recycle_state = (state->ID == id);
+        bool recycle_state = state->ID == id && !reinitialize;
         if (recycle_state)
```

Comment:
Both the `const` removal and the `!reinitialize` test seems unnecessary (the value is always false).